---
title: "To Do List"
date: '2019-2-10'
slug: todolist
---

# Plans

## Long-term goals

## Short-term to-do-list

### RNA-Seq pipeline
* Why pseudoalignment methods perform better
* Add the discussion  
  https://docs.google.com/document/d/15eWgezNB2gywotFjLMLqXF41e-8YM9K1lSpeOc00ZwI/edit
* Additional analysis?
* Website

### Methylation sequencing
* Evaluate CpG segments
  - Known methylation blocks
  - Known genome features
* Long-range genome-wide interaction
* Differential methylation analysis (Methylmix)
* Differential variance analysis
  - Barlett's test
  - Differential variance subgroups?
* Clustering
  - Consensus Clustering
  - WADP (Weighted average discrepant pairs)  
    https://www.nature.com/articles/35020115
* Visualization
* Wrap up a R package  
* Pan-Cancer analysis

### Single cell RNA-Seq analysis
* Single-cell AMARETTO on HNSC dataset
  - Pre-processing
      - Filter
      - Transformation
  - Data imputation
      - MAGIC
  - AMARETTO
      - Single-patient
      - Multiple-patient
* CROP-Seq dataset

### Collaboration
* Wei Dai, HKU

  Computational Framework Using Genomic and Transcriptomic Features for Predicting Resistance to PD-1/PDL-1 Checkpoint Inhibition in Gastrointestinal Tract (GIT) Cancer

  Objectives: Gastrointestinal tract (GIT) cancer is the leading cause for cancer mortality in Hong Kong. PD-1/PDL-1 inhibition has shown promising results in the treatment of advanced GIT cancer patients with tumor metastasis. However, identification of the patients who are highly responsive to the treatment is challenging. Our aim is to fully utilize the next-generation sequencing (NGS) data by developing an all-in-one web-based platform connected to a computational framework incorporating the benchmark tools to characterize multiple key genomic features and gene expression signatures relevant to the response to PD-1/PDL-1 inhibitors in GIT cancer.

  Design and Subjects: We will establish a computation framework to characterize the mutation load, mismatch repair (MMR) status, mutations of key genes, load of tumor neoantigens and HLA typing and expression of innate anti-PD-1 resistance signature (IPRES). The publicly available NGS data for colorectal cancer and esophageal squamous cell carcinoma will be collected and analyzed by this framework.

  Methods and Data Analysis: Multiple benchmark open-source bioinformatics tools will be incorporated into the computational framework, such as GATK, Mutect2, MSISeq, OptiType, NetMHC, and HT-seq. The sensitivity, specificity and positive prediction value for detecting the mutations, MMR deficiency and HLA typing will be evaluated by wet-lab experiments and/or dry-lab data.

  Expected Results: We will establish the baseline references for the selected key features using the public and in-house data. We expect that this tool is useful to facilitate the clinical trial design and guide the clinicians to tailor the treatment according to the genomic and transcriptomic profiles for precision medicine.

* Neural cell RNA-Seq (Nick Fisher)
* Ewing sarcoma and methylation, Helio Costa

### Fellowships
* Damon Runyon Cancer Research Foundation Postdoctoral Fellowship
